sys/vm/swap_pager.c

```diff
@@ -2221,11 +2220,10 @@ swp_pager_meta_cheri_get_tags(vm_page_t page)
 	modpi = page->pindex % SWAP_META_PAGES;
 	for (i = modpi * BITS_PER_TAGS_PER_PAGE;
 	    i < (modpi + 1) * BITS_PER_TAGS_PER_PAGE; i++) {
-		p = scan;
-		for (t = sb->swb_tags[i]; t != 0; t >>= j) {
-			j = ffsl((long)t);
-			cheri_restore_tag(p + j - 1);
-			p += j;
+		t = sb->swb_tags[i];
+		while ((j = ffsl((long)t) - 1) != -1) {
```

Comment:
The cast is there for signed-ness reasons, for some weird reason the functions take signed rather than unsigned inputs even though bitmaps should be unsigned...

Both forms get used in-tree. I think yours is a bit more common, though to me this way feels a bit more natural.

On 32-bit platforms we probably should have a 32-bit word for the mask, not uint64_t, the current design is a bit in not using u_long or similar.

Reply:
Oh, I would much prefer that we just used `u_long` instead of `uint64_t` for the tags.  Not sure how much it is worth it to do so or how invasive that change would be though.